python/src/spark_rapids_ml/umap.py

```diff
@@ -1441,19 +1423,39 @@ def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:
         return f"array<{pyspark_type}>"
 
     def write(self) -> MLWriter:
-        return _CumlModelWriterNumpy(self)
+        return _CumlModelWriterParquet(self)
 
     @classmethod
     def read(cls) -> MLReader:
-        return _CumlModelReaderNumpy(cls)
+        return _CumlModelReaderParquet(cls)
 
 
-class _CumlModelWriterNumpy(_CumlModelWriter):
+class _CumlModelWriterParquet(_CumlModelWriter):
     """
-    Override parent writer to save numpy objects of _CumlModel to the file
+    Override parent writer to write _CumlModel array attributes to Spark Dataframe as parquet
     """
 
     def saveImpl(self, path: str) -> None:
+
+        spark = _get_spark_session()
+
+        def write_sparse_array(array: scipy.sparse.spmatrix, df_dir: str) -> None:
+            indptr_df = spark.createDataFrame(array.indptr, schema=["indptr"])
+            indices_data_df = spark.createDataFrame(
+                pd.DataFrame({"indices": array.indices, "data": array.data})
```

Comment:
Better to add the row_id here in the pd dataframe in case it is shuffled when creating spark dataframe.